mod_form.php

```diff
@@ -641,8 +659,19 @@ public function data_preprocessing(&$defaultvalues) {
                 );
             }
         }
-    }
 
+        if ($config->defaulttrackingfields !== '') {
+            // Populate modedit form fields with previously saved values.
+            $defaulttrackingfields = clean_tracking_fields();
+            foreach ($defaulttrackingfields as $key => $defaulttrackingfield) {
+                $tfvalue = $DB->get_field('zoom_meeting_tracking_fields', 'value',
+                    array('meeting_id' => $defaultvalues['id'], 'tracking_field' => $key));
+                if (!empty($tfvalue)) {
+                    $defaultvalues[$key] = $tfvalue;
+                }
+            }
```

Comment:
It seems like this is doing several small queries. Maybe this would be more efficient?
```suggestion
            $defaulttrackingfields = zoom_clean_tracking_fields();
            $tfrows = $DB->get_records('zoom_meeting_tracking_fields', array('meeting_id' => $defaultvalues['id']));
            foreach ($tfrows as $tfrow) {
                $tfkey = $tfrow->tracking_field;
                if (!empty($defaulttrackingfields[$tfkey])) {
                    $defaultvalues[$tfkey] = $tfrow->value;
                }
            }
```

Reply:
Thanks for the fix - done